cluster.go

```diff
@@ -192,28 +199,40 @@ func (n *clusterNode) Close() error {
 	return n.Client.Close()
 }
 
+type CNode interface {
+	Client() *Client
+}
+
+// latency funcs return true if alls well, false if they should be marked as failing
+func defaultLatencyFunc(c *Client) bool {
+	cmd := c.Ping(context.Background())
+	if isLoadingError(cmd.err) {
+		return false
+	}
+	return true
+}
+
 func (n *clusterNode) updateLatency() {
 	const numProbe = 10
 	var dur uint64
 
-	var i int
-	for i = 0; i < numProbe; i++ {
+	for i := 0; i < numProbe; i++ {
 		time.Sleep(time.Duration(10+rand.Intn(10)) * time.Millisecond)
 
 		start := time.Now()
-		cmd := n.Client.Ping(context.TODO())
-		// sometimes failures can be really fast and actually make them the fastest node
-		if cmd.err != nil {
-			if isLoadingError(cmd.err) || isConnectionError(cmd.err) {
-				internal.Logger.Printf(context.Background(), "latency marking as failing %s %s", n.String(), cmd.err)
-				n.MarkAsFailing()
-				break
-			}
+		latencyHealthFunc := defaultLatencyFunc
+		if n.Client.Options().LatencyHealthFunc != nil {
+			latencyHealthFunc = n.Client.Options().LatencyHealthFunc
+		}
+		if !latencyHealthFunc(n.Client) {
+			n.MarkAsFailing()
+			atomic.StoreUint32(&n.latency, math.MaxUint32) // turn latency way up
```

Comment:
Why is the magic latency value needed when there is already a failing flag?  Not sure they'll accept this upstream as it isn't really backwards compatible.

Reply:
This is the latency a new node gets when it first shows up and hasn't had a latency func yet. If it fails it seems only natural to do that. Additionally another health check would be run before the failure timeout has occurred so if it passes it will have a valid non-max int latency.

I will make the maxuint32 be a constant for default latency and use it in both places.